olive/model.py

```diff
@@ -104,6 +107,25 @@ def prepare_session(
         """
         raise NotImplementedError()
 
+    def download_from_azureml(self, azureml_client: AzureMLClient, download_path: Union[Path, str]):
```

Comment:
>azureml_client [](http://example.com/codeflow?start=36&length=14)

use MLClient?

Reply:
AzureMLClient is a wrapper here for MLClient. Otherwise if we specify MLClient type hint we will need to import MLClient to model.py

Reply:
We could use "MLClient" as the type hints.

Reply:
Yeah that's what I mean. MLClient is from azure ai package "from azure.ai.ml import MLClient", we don't want to import this in model module level.

Reply:
we need not import the MLClient. We could either use string or just leave the type hints empty here

Reply:
I don't like a placeholder string or empty type hint here. Any reason that model download should not know AzureMLClient?

Reply:
Because model will have knowledge about the implementation detail of AzureMLClient. This is a design flaw we need avoid.

Reply:
[TYPE_CHECKING](https://docs.python.org/3/library/typing.html#typing.TYPE_CHECKING) should be the thing your are looking for.